store/driver/txn/txn_driver.go

```diff
@@ -288,5 +289,10 @@ type TiDBKVFilter struct{}
 
 // IsUnnecessaryKeyValue defines which kinds of KV pairs from TiDB needn't be committed.
 func (f TiDBKVFilter) IsUnnecessaryKeyValue(key, value []byte, flags tikvstore.KeyFlags) bool {
-	return tablecodec.IsUntouchedIndexKValue(key, value)
+	isUntouchedValue := tablecodec.IsUntouchedIndexKValue(key, value)
+	if isUntouchedValue && flags.HasPresumeKeyNotExists() {
+		panic(fmt.Sprintf("unexpected path the untouched key=%s value=%s contains PresumeKeyNotExists flag keyFlags=%v",
```

Comment:
Panic may be too strict. How about making `KVFilter.IsUnnecessaryKeyValue()` return `(bool, error)`? 
https://github.com/tikv/client-go/blob/c0e876615440ebe603550f7bdcfe750ebe90dd5d/txnkv/transaction/2pc.go#L382-L385

If error is not nil, `twoPhaseCommitter.initKeysAndMutations()` reports it to user.